File: Results/DER/Translation/translation_output/w_test/gpt-4/avatar/java/python/atcoder_ABC149_C.py
The provided Java code finds the smallest prime number that is greater than or equal to the given input number. Below is the translated code in Python:

```python
def is_prime(num):
    if num <= 1:
        return False
    for i in range(2, int(num ** 0.5) + 1):
        if num % i == 0:
            return False
    return True

def next_prime(X):
    prime = X
    while True:
        if is_prime(prime):
            return prime
        prime += 1

# Test input
X = 9552

# Expected output
print(next_prime(X))
```

**Explanation:**
- The `is_prime` function checks if a given number is prime.
- The `next_prime` function finds the next prime number that is greater than or equal to the given input.
- The test input `X = 9552` should produce the expected output `9587` when the `next_prime` function is called.

File: Results/DER/Translation/translation_output/w_test/gpt-4/avatar/python/java/atcoder_ABC149_C.java
Below is the translation of the provided Python code into Java. I have wrapped the code in an appropriate class and method structure. A main method reads the user's input and then calls another method that performs the same logic as the original Python code.

```java
import java.util.Scanner;

public class PrimeFinder {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();

        while (!isPrime(N)) {
            N++;
        }

        System.out.println(N);
        scanner.close();
    }

    public static boolean isPrime(int n) {
        if (n == 2) {
            return true;
        }
        for (int x = 2; x <= Math.sqrt(n) + 1; x++) {
            if (n % x == 0) {
                return false;
            }
        }
        return true;
    }
}
```

This Java code should produce the same output as the original Python code when given the test input of `9552`. It will find the next prime number, which is `9587`.
